### DISCUSSION TITLE
cannot save MLP_model

### DISCUSSION THREAD

**Discussion by beinathan** [online-ml/river#902] [Q&A] [answered]
When I try to use pickle to save mlp_model，it has  the following error:
**AttributeError: Can't pickle local object 'MLP.__init__.<locals>.<lambda>'**
I also has tried dill, but still not, so how to solve it?

**Comment by MaxHalford** [comment, marked as answer]
Hello. This is because there's a lambda function in the `MLP` class. I'll send a fix which will be available in the next release. Thanks for notifying us!

**Reply by tamoghnabh** [reply]
Hey Max, I too am facing the issue, I have installed the development version, still not working. Any fix?

**Reply by MaxHalford** [reply]
Hey! Can you share some details about your setup? Your Python version, numpy, etc.? I'm going to dive into this issue before making the next release.

**Reply by tamoghnabh** [reply]
[setup.txt](https://github.com/online-ml/river/files/8778565/setup.txt)
Hey Max! the file contains all the packages installed with River in my environment. Fyi, I had to design the environment carefully. I require GPFlow and River in the same environment and I have noticed that slight tweak here and there breaks the environment and River fails to load with some error with Numpy. Hoping to get a fix soon.

**Comment by beinathan** [comment]
Okay, thanks for your answer.

**Reply by MaxHalford** [reply]
You can install the development version if you want :)

**Reply by beinathan** [reply]
Thanks so much, I have installed it!
